trin-core/src/portalnet/overlay.rs

```diff
@@ -144,14 +144,11 @@ impl<TContentKey: OverlayContentKey + Send> OverlayProtocol<TContentKey> {
         &self,
         talk_request: &TalkRequest,
     ) -> Result<Response, OverlayRequestError> {
-        let request = match Message::from_bytes(talk_request.body()) {
-            Ok(Message::Request(request)) => request,
-            Ok(_) => {
-                return Err(OverlayRequestError::InvalidRequest(format!(
-                    "Message not recognized request type: {:?}",
-                    talk_request.body()
-                )))
-            }
+        let request = match Message::try_from(Vec::<u8>::from(talk_request.body())) {
```

Comment:
Would it make sense to implement `TryFrom<TalkRequest>` as well as `TryFrom<Vec<u8>>` on `Message` .... or maybe even `impl TryFrom<TalkRequest> for Request`?

Reply:
For a similar reason mentioned below, we cannot implement `TryFrom<TalkRequest>` for `Request`. The `Request` type does not derive the SSZ encoding traits, so it cannot handle an arbitrary byte array.

We could implement `TryFrom<TalkRequest>` for `Message`, but I don't see much additional utility here.